src/main/java/spoon/reflect/visitor/DefaultJavaPrettyPrinter.java

```diff
@@ -2359,4 +2360,22 @@ public void visitCtRecordPattern(CtRecordPattern recordPattern) {
 		elementPrinterHelper.printList(recordPattern.getPatternList(),
 			null, false, "(", false, false, ",", true, false, ")", this::scan);
 	}
+
+	@Override
+	public void visitCtReceiverParameter(CtReceiverParameter receiverParameter) {
+		elementPrinterHelper.writeComment(receiverParameter);
+		elementPrinterHelper.writeAnnotations(receiverParameter);
+
+		printer.writeIdentifier(receiverParameter.getType().getSimpleName());
+		printer.writeSpace();
+		boolean isInnerClass = receiverParameter.getType().getTopLevelType().equals(receiverParameter.getParent(CtType.class));
```

Comment:
According to Qodana, this is wrong, and I think it's also wrong when having nested classes in nested classes